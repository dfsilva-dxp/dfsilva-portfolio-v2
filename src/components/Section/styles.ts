import styled, { css } from "styled-components";

export const Content = styled.section`
  ${({ theme }) => css`
    width: 100%;
    height: fit-content;
    border-bottom: 1px solid ${theme.colors.glitch};
  `}
`;
