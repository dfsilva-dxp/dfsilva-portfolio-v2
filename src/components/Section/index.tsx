import { ISectionProps } from "./section";

import * as S from "./styles";

const Section = ({ children }: ISectionProps) => {
  return <S.Content>{children}</S.Content>;
};

export default Section;
