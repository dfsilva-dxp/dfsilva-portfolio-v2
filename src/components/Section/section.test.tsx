import { screen } from "@testing-library/dom";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Section from ".";

describe("<Section />", () => {
  it("should render a children", () => {
    const { container } = renderWithTheme(
      <Section>
        <h1>Heading</h1>
      </Section>
    );

    expect(
      screen.getByRole("heading", { name: /heading/i })
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
