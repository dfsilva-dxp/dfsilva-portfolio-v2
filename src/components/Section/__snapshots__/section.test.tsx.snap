// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render a children 1`] = `
.c0 {
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  border-bottom: 1px solid #f2f2f2;
}

<div>
  <section
    class="c0"
  >
    <h1>
      Heading
    </h1>
  </section>
</div>
`;
